components/eventing-controller/pkg/handlers/eventtype/clean.go

```diff
@@ -22,23 +30,27 @@ func NewCleaner(eventTypePrefix string, applicationLister *application.Lister, l
 	return &cleaner{eventTypePrefix: eventTypePrefix, applicationLister: applicationLister, logger: logger}
 }
 
-// Clean cleans the application name segment in the event-type from none-alphanumeric characters
-// and returns the clean event-type, or returns an error if the event-type parsing failed
+// Clean cleans the event-type from none-alphanumeric characters and returns it
+// or returns an error if the event-type parsing failed
 func (c cleaner) Clean(eventType string) (string, error) {
 	appName, event, version, err := parse(eventType, c.eventTypePrefix)
 	if err != nil {
 		c.logger.Error(err, "failed to parse event-type", "prefix", c.eventTypePrefix, "type", eventType)
 		return "", err
 	}
 
-	// handle existing applications
-	if appObj, err := c.applicationLister.Get(appName); err == nil {
-		eventType = build(c.eventTypePrefix, application.GetCleanTypeOrName(appObj), event, version)
-		return eventType, nil
+	// clean the application name
+	var eventTypeClean string
+	if appObj, err := c.applicationLister.Get(appName); err != nil {
+		c.logger.Info("cannot find application", "name", appName)
+		eventTypeClean = build(c.eventTypePrefix, application.GetCleanName(appName), event, version)
+	} else {
+		eventTypeClean = build(c.eventTypePrefix, application.GetCleanTypeOrName(appObj), event, version)
 	}
 
-	// handle non-existing applications
-	c.logger.Info("failed to get application", "name", appName)
-	eventType = build(c.eventTypePrefix, application.GetCleanName(appName), event, version)
-	return eventType, nil
+	// clean the event-type segments
+	eventTypeClean = invalidEventTypeSegment.ReplaceAllString(eventTypeClean, "")
```

Comment:
Would hide the usage of the regexp inside a function `CleanEventType()` instead of using it directly. So intention would be more clear.

Reply:
Done.